core/bin/external_node/src/main.rs

```diff
@@ -756,7 +768,26 @@ async fn main() -> anyhow::Result<()> {
     // Initial setup.
     let opt = Cli::parse();
 
-    let mut config = ExternalNodeConfig::new().context("Failed to load node configuration")?;
+    let mut config = if let Some(config_path) = opt.config_path.clone() {
+        let secrets_path = opt
+            .secrets_path
+            .clone()
+            .context("File based config can't be used partially please provide all paths")?;
```

Comment:
IIUC, this relationship can be enforced on the `clap` level using [`requires`](https://docs.rs/clap/latest/clap/struct.Arg.html#method.requires) attribute.